h/services/search_index/_queue.py

```diff
@@ -110,6 +111,18 @@ def add_by_user(self, userid, tag, force=False, schedule_in=None):
         where = [Annotation.userid == userid]
         self.add_where(where, tag, Queue.Priority.SINGLE_USER, force, schedule_in)
 
+    def add_by_group(self, groupid, tag, force=False, schedule_in=None):
```

Comment:
The `force` and `schedule_in` arguments were added because the `add_by_user` method above uses them. However they are never actually used. Should I keep them?

Reply:
I think you may want `force=True` though, not sure.

With `force=False` it'll read the annotation from both Postgres and Elasticsearch and [compare only their updated time and userid](https://github.com/hypothesis/h/blob/d3304f1b6c9214429f93f5459fd758008835c0dc/h/services/search_index/_queue.py#L265-L271) and reindex the annotation only if those fields differ.

Whereas with `force=True` it'll just blindly reindex all the user's annotations.

I'm not sure what'll happen in the case of annotations that were deleted from the DB but have now been added back to the DB. They would have been marked as deleted in Elasticsearch, but still be present in Elasticsearch. I think it fully tombstones the Elasticsearch document, so it should actually now be lacking the `updated` and `user` fields and so should compare as different from the DB record and get reindexed even without `force`.

There's certainly nothing in [the Elasticsearch query it uses](https://github.com/hypothesis/h/blob/d3304f1b6c9214429f93f5459fd758008835c0dc/h/services/search_index/_queue.py#L248-L263) that looks like it filters out deleted annotations.

`force=True` is faster

Reply:
I've kept the `force` and `schedule_in` options for the moment for consistency between the methods for indexing users and groups. We can easily remove them later.